#159989013 User can reset password

anneKay opened the pull request:
#### What does this PR do?

- User can receive email with a link to reset password
- User can reset password by clicking on the link sent

#### Description of Task to be completed?

    - Add redux functionality
    - Add scss style for password components
    - Add redirection to loginpage after password reset
    - Display error messages for bad requests
    - Write tests

#### How should this be manually tested?
- `git clone https://github.com/andela/fargo-ah-client.git Authors-Haven-Client`
- `npm install`
- `npm run start` the start script
- Go to '/login' path and click on `login` or `register`
- Click on password reset
- Enter valid email 
- Click on link sent through email
- Enter password to reset


#### Any background context you want to provide?

#### What are the relevant pivotal tracker stories?

- [#159989014](https://www.pivotaltracker.com/story/show/159989013)

#### Screenshots (if appropriate)

![image](https://user-images.githubusercontent.com/24938970/45618199-dd7d2280-ba6c-11e8-83bd-d1ad4e280ad8.png)

![image](https://user-images.githubusercontent.com/24938970/45618217-ebcb3e80-ba6c-11e8-8bc7-665296e90206.png)

![image](https://user-images.githubusercontent.com/24938970/45618296-20d79100-ba6d-11e8-9383-674ba8cf3d52.png)


#### Questions:

N/A

veeqtor committed:
ft(form-modals): Create the login and register modal component

- Write test
- Create the login modal component
- Create the register modal component
- Create a style for the forms in the modals
- Modifiy the header card component
- Write unit tests

[Delivers #160259705]

ascii-dev commented on src/components/forms/PasswordResetForm.js:
```diff
@@ -0,0 +1,86 @@
+    }
+  };
+
+  validate = (data) => {
```
Fixed this???

ascii-dev commented on src/styles/reset.scss:
```diff
@@ -0,0 +1,166 @@
+      }
+    }
+  }
+}
```
New line please?

ascii-dev commented on src/styles/resetSuccess.scss:
```diff
@@ -0,0 +1,59 @@
+      }
+   }
+  }
+}
```
New line 😉

ascii-dev reviewed (changes requested):
Kindly fix HoundCI issues and also I think you should add the base endpoint url to env, so that it won't be hard coded every where it's needed just incase of when we deploy. Thanks.

Oloyedesinmiloluwa commented on src/redux/actions/resetPassword.js:
```diff
@@ -0,0 +1,54 @@
+  RESET_PASSWORD_SUCCESS, RESET_PASSWORD_FAILURE,
+  UPDATE_PASSWORD_FAILURE, UPDATE_PASSWORD_SUCCESS,
+} = actions;
+const BASE_URL_1 = 'https://fargo-ah-staging.herokuapp.com/api';
```
well done, I think handling this base url in a dynamic way is just appropriate, what about using an environment variable for this?

anneKay replied:
Thank you Sinmi for your suggestion, while adding the base URL to env file is a good option that the team is trying to resolve, I suggest we add the base URL to Axios using `axios.defaults.baseURL`

Oloyedesinmiloluwa reviewed (approved):
good

anneKay committed:
ft(reset): reset user password

- write unit tests
- user email component
- send user email to reset password
- user password component
- update user password

[Deliveres #59989013]

anuonifade reviewed (approved):
LGTM
